### ISSUE TITLE
Interface(s) of a service is always registered as `Transient`.

### ISSUE THREAD

**Issue by tchelidze** [Particular/NServiceBus.Extensions.DependencyInjection#20]
[ServiceCollectionAdapter.RegisterInterfaces](https://github.com/Particular/NServiceBus.Extensions.DependencyInjection/blob/master/src/NServiceBus.Extensions.DependencyInjection/ServiceCollectionAdapter.cs#L76) method registers interface of a target implementation type with `Transient` lifetime, instead of the lifetime of implementation type, that causes `ComponentActivatorException`  for `Castle.Windsor` container, when that interface is resolved **second times**

```
Castle.MicroKernel.ComponentActivator.ComponentActivatorException: 
Instance {ImplementationType} of component {IInterface}_affce5c5-40e5-4a45-96aa-144328d7346e is already being tracked.
The factory method providing instances of the component is reusing instances, but the lifestyle of the component is Transient which requires new instance each time.
In most cases it is advised for the factory method not to be handling reuse of the instances, but to chose a lifestyle that does that appropriately.
Alternatively, if you do not wish for Windsor to track the objects coming from the factory change your registration to '.UsingFactoryMethod(yourFactory, managedExternally: true)''
```
As an example, [BlobStorageDataBus](https://github.com/Particular/NServiceBus.DataBus.AzureBlobStorage/blob/master/src/DataBus/BlobStorageDataBus.cs) which implements `IDataBus` is [registered as a singleton](https://github.com/Particular/NServiceBus.DataBus.AzureBlobStorage/blob/master/src/DataBus/AzureDataBusPersistence.cs#L29) while `IDataBus` ends up being registered as a `Transient`, causing above error when `IDataBus` is resolved second times.

**Comment by timbussmann** [comment]
Thanks for raising this. That Castle.Windsor behavior hasn't been taken into account. The intention of the transient registration was to forward the request to the underlying "main" registration which does use the right scope but it seems Windsor explicitly tries to detect such use cases.

**Comment by tchelidze** [comment]
@timbussmann true. Besides, it is will also cause an issue when you try to inject that interface into a singleton service (while in fact it should be perfectly possible to inject singleton into a singleton, but since an interface is transient it is no longer possible).

**Comment by timbussmann** [comment]
I've been trying to reproduce this error without using the BlobStorageDataBus (in which case I get the same error) but with no success so far. Have you been observing this error on other configurations/registrations @tchelidze ?

**Comment by tchelidze** [comment]
Nope, but that's probably because no other service is resolved from DI container more than once @timbussmann  ?

**Comment by timbussmann** [comment]
weirdly, even if I try to explicitly resolve a service multiple times I don't run into any issues.

**Comment by timbussmann** [comment]
ah, the instance tracking only seems to be an issue when using types implementing `IDisposable`. I have been able to reproduce this outside the data bus with that.

**Comment by timbussmann** [comment]
This issue has been fixed and released with release 1.0.1 of NServiceBus.Extensions.DependencyInjection.
